spec/features/trains/admin_can_create_train_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+feature 'admin can create train' do
+  let(:response_double) { double }
+  let(:admin) { create(:admin, email: 'email@email.com', password: 'password') }
+
+  background do
+    expect_any_instance_of(TrainApi).to receive(:index).and_return([])
+    expect_any_instance_of(TrainApi).to receive(:create).and_return(response_double)
+    expect(response_double).to receive(:success?).and_return(true)
+    expect(response_double).to receive(:[]).and_return("new_name successfully created")
+    log_in_as(admin)
+    visit(trains_path)
+    allow_any_instance_of(TrainApi).to receive(:index).and_return([])
+    click_link('New train')
```

Comment:
can you actually visit this page?
i dont see a `def new` on your controller for rendering this page

Reply:
turns out you can, but its weird rails magic that we shouldnt rely on.

Reply:
> i dont see a `def new` on your controller for rendering this page

do i add `self.class.get('/v1/trains/new')` into the services part for the `def new`? 
because there isnt a new path for the api

Reply:
This app. not the api which is a different app
requires its actions and views to be defined.
you need a NEW action on THIS app (NOT the api) so you can render the NEW view.
rails will automagically do this for you if you have the route, and the views but we want to be explicit.
so we need a NEW action in out TRUCK APP (NOT train API) to have the new action for the train controller so we can show we are explicitly rendering that view